hyperspy/component.py

```diff
@@ -1324,3 +1326,43 @@ def on_active_change(change):
                              ' a notebook')
             else:
                 raise
+    @property
+    def is_linear(self):
+        "Loops through the component's free parameters, checks that they are linear"
+        linear = True
+        if len(self.free_parameters) > 1:
+            return False
+        for para in self.free_parameters:
+            if not para._is_linear:
+                linear = False
+        return linear
+    
+    @property
+    def independent_parameters(self):
+        """Lists all parameters that are independent"""
+        return [para for para in self.parameters if para._is_independent]
+
+    @property
+    def independent_term(self):
+        "Get value of the independent term of the component at current indices. Returns 0 for most components."
+        return self._independent_term()
+
+    def _independent_term(self):
+        return 0
+
+    def _check_only_one_linear_parameter(self):
```

Comment:
Isn't this function unnecessary? The component can't be linear if the number of free parameters is > 1.